packages/simulation/src/convert.ts

```diff
@@ -0,0 +1,647 @@
+  const keySet = new Set<string>()
+  let newKeyAdded = true
+  let tempFromState = 0
+
+  // This while loop ensures that once states are merged, their to states are merged and checked to see if they exist. If they all do, then there are no new combinations to be made and just exit, otherwise
+  // the states need to be merged
+  while (newKeyAdded) {
+    newKeyAdded = false
+
+    for (const symbol of Object.keys(symbolToStatesMap)) {
+      symbolToStatesMap[symbol].forEach(([curTransitionElem]) => {
+        const [curKeyCombination, combinedState] = combineStates(initialTransitionTable, symbolsArray, curTransitionElem, symbolToStatesMap, symbol)
+        if (combinedState) {
+          if (!keySet.has(curKeyCombination)) {
+            newCombinedStates[curKeyCombination] = combinedState
+            keySet.add(curKeyCombination)
+            newKeyAdded = true
+          }
+        }
+      })
+    }
+    symbolToStatesMap = {}
+    // The from state is just temporary and is not important, so use to identify the sequences
+    tempFromState = 0
+    for (const [key, value] of Object.entries(newCombinedStates)) {
+      for (let transElem = 0; transElem < value.length; transElem++) {
+        if (newCombinedStates[key][transElem].symbol in symbolToStatesMap) {
+          for (let toStateElem = 0; toStateElem < newCombinedStates[key][transElem].tostates.length; toStateElem++) {
+            symbolToStatesMap[newCombinedStates[key][transElem].symbol].push([tempFromState, newCombinedStates[key][transElem].tostates[toStateElem]])
+          }
+          tempFromState++
+        } else {
+          // Equal just one state, then push the rest
+          symbolToStatesMap[newCombinedStates[key][transElem].symbol] = [[tempFromState, newCombinedStates[key][transElem].tostates[0]]]
+          for (let toStateElem = 1; toStateElem < newCombinedStates[key][transElem].tostates.length; toStateElem++) {
+            symbolToStatesMap[newCombinedStates[key][transElem].symbol].push([tempFromState, newCombinedStates[key][transElem].tostates[toStateElem]])
+          }
+          tempFromState++
+        }
+      }
+    }
+  }
+
+  // Step 3.5, where the merged states will be added to the initialTransitionTable and current transitions will be modified accordingly. Everything below will do just that
+  // by using reference points to the first combined state
+  const lastKey = Object.keys(initialTransitionTable)[Object.keys(initialTransitionTable).length - 1]
+  let lastKeyNumber = parseInt(lastKey)
+  const lastKeyArray = []
+  // Increase by one to get to next key (the new key for initial table)
+  lastKeyNumber += 1
+  for (let mergedElem = 0; mergedElem < Object.keys(newCombinedStates).length; mergedElem++) {
+    const stateArray = Object.keys(newCombinedStates)[mergedElem].split(',').map(Number)
+    for (let curState = 0; curState < numberOfNFAStates; curState++) {
+      if (initialTransitionTable[curState] === undefined) {
+        continue
```

Comment:
Just have it as `if (initialTransitionTable[curState] !== undefined)` instead of this `if/else`

Reply:
I notice this idiom is used in a few places, make sure to clean it up in the other places